servlet-core/src/main/java/io/micronaut/servlet/http/ServletHttpHandler.java

```diff
@@ -581,59 +600,57 @@ private void subscribeToResponsePublisher(HttpRequest<Object> req,
+                                emitter.success();
                             }
                         }
                     }));
                 } else {
-                    result = Single.fromFuture((Future<?>) result);
+                    result = Mono.fromCallable(((Future<?>) result)::get);
                 }
             }
 
+            Argument<?> firstArg = genericReturnType.getFirstTypeVariable().orElse(null);
             if (firstArg != null && HttpResponse.class.isAssignableFrom(firstArg.getType()) && Publishers.isConvertibleToPublisher(result)) {
                 //noinspection unchecked
-                return Publishers.convertPublisher(result, Flowable.class);
+                return Publishers.convertPublisher(result, Flux.class);
             } else {
                 if (isSuspended) {
                     boolean isKotlinFunctionReturnTypeUnit =
                             route instanceof MethodBasedRouteMatch &&
                                     isKotlinFunctionReturnTypeUnit(((MethodBasedRouteMatch) route).getExecutableMethod());
                     final Supplier<CompletableFuture<?>> supplier = ContinuationArgumentBinder.extractContinuationCompletableFutureSupplier(req);
                     if (isKotlinCoroutineSuspended(result)) {
-                        return Flowable.create(emitter -> {
+                        return Flux.create((FluxSink<MutableHttpResponse<?>> emitter) -> {
                             CompletableFuture<?> f = supplier.get();
                             f.whenComplete((o, throwable) -> {
                                 if (throwable != null) {
-                                    emitter.onError(throwable);
+                                    emitter.error(throwable);
                                 } else {
                                     if (o == null) {
-                                        emitter.onNext(httpResponse.status(HttpStatus.NOT_FOUND));
+                                        emitter.next(httpResponse.status(HttpStatus.NOT_FOUND));
                                     } else {
                                         if (!isKotlinFunctionReturnTypeUnit) {
                                             httpResponse.body(o);
                                         }
-                                        emitter.onNext(httpResponse);
+                                        emitter.next(httpResponse);
                                     }
-                                    emitter.onComplete();
+                                    emitter.complete();
                                 }
                             });
-                        }, BackpressureStrategy.ERROR);
+                        }, FluxSink.OverflowStrategy.ERROR);
                     } else {
-                        Object suspendedBody;
-                        if (isKotlinFunctionReturnTypeUnit) {
-                            suspendedBody = Completable.complete();
```

Comment:
We probably need to replace `Completeable` with `Mono<Void>`.

Reply:
Easiest way (I think?) to get a `Mono<Void>` is `Mono.empty().then()`.